pkg/cmd/cli/cmd/set/env.go

```diff
@@ -200,8 +202,23 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 			return nil
 		})
 		if !ok {
-			skipped++
-			continue
+			ok, err = f.UpdateObjectEnvironment(info.Object, func(vars *[]kapi.EnvVar) error {
```

Comment:
maybe I should add comment saying that this is a fallback function for things that don't have pod spec.